controllers/machinehealthcheck_controller_test.go

```diff
@@ -256,19 +278,22 @@ var _ = Describe("MachineHealthCheck", func() {
 			})
 		})
 
+		// to be fixed!!
 		Context("it should ensure an owner reference is present", func() {
 			Specify("when no existing ones exist", func() {
 				mhc.Spec.ClusterName = cluster.Name
 				mhc.OwnerReferences = nil
+				mhc.SetGenerateName("owner-reference-when-no-existing-ones-exist-")
 				Expect(testEnv.Create(ctx, mhc)).To(Succeed())
-
+				objectKey := util.ObjectKey(mhc)
 				Eventually(func() []metav1.OwnerReference {
-					err := testEnv.Get(ctx, util.ObjectKey(mhc), mhc)
+					err := testEnv.Get(ctx, objectKey, mhc)
```

Comment:
I'm not sure this change is strictly necessary?

Reply:
Yes. It's not strictly necessary